packages/arb-bridge-eth/contracts/bridge/SequencerInbox.sol

```diff
@@ -187,6 +188,44 @@ contract SequencerInbox is ISequencerInbox, Cloneable {
         );
     }
 
+    function addSequencerL2BatchFromOriginWithGasRefunder(
+        bytes calldata transactions,
+        uint256[] calldata lengths,
+        uint256[] calldata sectionsMetadata,
+        bytes32 afterAcc,
+        IGasRefunder gasRefunder
+    ) external {
+        // solhint-disable-next-line avoid-tx-origin
+        require(msg.sender == tx.origin, "origin only");
+
+        // Add in a lower bound of calldata cost to the gas estimate
+        // startGasLeft = gasleft() + 4 * calldatasize()
+        uint256 startGasLeft;
+        assembly {
+            startGasLeft := mul(calldatasize(), 4)
+        }
+        startGasLeft += gasleft();
+
+        uint256 startNum = messageCount;
+        bytes32 beforeAcc = addSequencerL2BatchImpl(
+            transactions,
+            lengths,
+            sectionsMetadata,
+            afterAcc
+        );
+        emit SequencerBatchDeliveredFromOrigin(
+            startNum,
+            beforeAcc,
+            messageCount,
+            afterAcc,
+            inboxAccs.length - 1
+        );
+
+        if (gasRefunder != IGasRefunder(0)) {
+            gasRefunder.onGasSpent(msg.sender, startGasLeft - gasleft());
```

Comment:
Should we do a contract existence check before the external call?
If this accidentally calls an EOA, I believe we won't see a revert

Reply:
Should we add a try catch in case this external call reverts?
Shouldn't be an issue since the `gasRefunder` is trusted code and should only hit a revert if badly configured